### ISSUE TITLE
Support for Let's Encrypt ACMEv2 and Wildcard certificate

### ISSUE THREAD

**Issue by Dibbyo456** [lesaff/serverpilot-letsencrypt#65]
Discussion - https://community.letsencrypt.org/t/acmev2-and-wildcard-launch-delay/53654

The API will be release tomorrow (February 27) publicly. I think @lesaff will have to revamp the script to make it support, right?

since, we don't have to generate SSL for every subdomain now, only the for root domain.

**Comment by caroseuk** [comment]
I dont think so, if that was the case, anyone who uses LetsEncrypt (Millions of users) would need to change scripts/update systems.

TLS-SNI01 has been disabled due to the security flaw, and it's fallback is the HTTP-01 method. so existing certs should still renew without issue.

There will eventually be an update to the Cerbot/LetsEncrypt upstream packages which will more than likely have an added flag or similar for the wildcard functionality. 

In short, I don't think the script will need amending immediately, though would be cool, once the packages are updated :)

**Comment by Dibbyo456** [comment]
Yeah, that’s exactly I meant.
The script will need a update so we will have to generate certificate for only root domain, but the certificate will work any subdomain.
That’s exactly what I want. I hate to generate certificate for each subdomain.

**Comment by lesaff** [comment]
Yes, this will require a rewrite of the bash script. I am also working on a new Node-based script to replace the old one, complete with web interface. Stay tuned.

**Comment by caroseuk** [comment]
@lesaff have you got a repo I can play with? ;)

**Comment by lesaff** [comment]
Not yet. I will put it on a new branch when it's ready for testing. I will definitely need your help.

**Comment by Dibbyo456** [comment]
@lesaff The new API seems to have released. Any updates?

**Comment by hbakhtiyor** [comment]
@lesaff any updates? only one branch in the repo
